src/main.zig

```diff
@@ -729,21 +729,51 @@ pub fn main() void { // MARK: main()
 	}
 }
 
+const maxRecursionDepth = 128;
+const typeIdSentinel = std.math.maxInt(usize);
+
 /// std.testing.refAllDeclsRecursive, but ignores C imports (by name)
 pub fn refAllDeclsRecursiveExceptCImports(comptime T: type) void {
+	var visited: [maxRecursionDepth]usize = undefined;
+	@memset(&visited, typeIdSentinel);
+	_refAllDeclsRecursiveExceptCImports(T, &visited, 0);
+}
+
+fn _refAllDeclsRecursiveExceptCImports(comptime T: type, visited: *[maxRecursionDepth]usize, index: usize) void {
 	if(!@import("builtin").is_test) return;
+	if(index >= maxRecursionDepth) {
+		std.debug.print("Reached recursion limit of {d} while visiting type {s}\n", .{maxRecursionDepth, @typeName(T)});
+		return;
+	}
+	const typeId = utils.typeId(T);
+
+	for(visited) |id| {
+		if(id == typeId) {
+			std.debug.print("Detected cycle at type {s} ({d})\n", .{@typeName(T), typeId});
+			return;
+		}
+		if(id == typeIdSentinel) {
+			break;
+		}
+	}
+
+	std.debug.print("Visiting type {s} ({d}) (depth: {d})\n", .{@typeName(T), typeId, index});
```

Comment:
```suggestion
```

Reply:
I don't think this really helps.
I would prefer a test in the CI that globally checks that all files were imported publicly in a path from main. But that's out of scope for this PR.

Reply:
Well, that helps in a sense that it prevents infinite cycles. You would have to remove recursive traversal completely to remove this problem and you would need to import literally everything, including commands, rotations etc. into main to guarantee it can all be checked as flat structure.

Reply:
Well, that is helpful in determining where the hell something went wrong during visiting stuff, as you know what you have visited before so you have a traceback.